src/components/post-card/PostCard.js

```diff
@@ -1,64 +1,78 @@
+        const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+        // const renderImage = () => {
+        //     return hasImage ? <img src={kittyUrl} alt="cat"/> : <img src={DefaultImg} alt="DefaultImg"/>;
+        // };
 
-  const renderImage = () => {
-    return hasImage ? (
-      <img src={kittyUrl} />
-    ) : (
-      <img src={DefaultImg} />
-    );
-  };
+        return (
+            <div className="may-post-card card">
 
-  return (
-    <div className="may-post-card card">
-      {hasImage && (
-        <div className="may-post-card-img" id="my-block" onClick={() => {
-          alert('ghvcdhfvbdfsjvbdf');
-        }}>
-          <img src={kittyUrl} />
-        </div>
-      )
-      }
-      {!hasImage && (
-        <div className="may-post-card-img">
-          <img src={DefaultImg} />
-        </div>
-      )}
+                <button
+                    onClick={this.onShowHideClick}
+                    type="button"
+                    className="btn btn-outline-secondary ">
+                    hide/show comment
+                </button>
+                {
+                    showComments && (<div>
 
-      <div className="card-body">
-        <h4 className="card-title title">{title}</h4>
-        <div className="card-text body">
-          {body}
-        </div>
-        <blockquote className="blockquote">
-          <footer className="blockquote-footer">Author:
-            <cite title="Source Title">{author}</cite>
-          </footer>
-        </blockquote>
-      </div>
+                        {hasImage && (<div className="may-post-card-img" id="my-block"
+                                           onClick={() => alert('You clicked on the image')}>
+                            <img src={kittyUrl} alt="cat"/>
+                        </div>)
+                        }
 
-      {!!comments.length && <label>Comments:</label> }
-      {
-        comments.map(comment => (<Comment comment={comment} key={comment.id}/>))
-      }
-    </div>
-  );
+                        {!hasImage && (<div className="may-post-card-img">
+                            <img src={DefaultImg} alt="DefaultImg"/>
+                        </div>)
+                        }
+
+                        <div className="card-body">
+                            <h4 className="card-title title">{title}</h4>
+                            <div className="card-text body">{body}</div>
+                            <blockquote className="blockquote">
+                                <footer className="blockquote-footer">Author:
+                                    <cite title="Source Title">{author}</cite>
+                                </footer>
+                            </blockquote>
+                        </div>
+
+                        {!!comments.length && <label>Comments:</label>}
+                        {
+                            comments.map(comment => (<Comment comment={comment} key={comment.id}/>))
```

Comment:
And here should be check: is showComments = true then we render comments, otherwise - no

Reply:
Fixed it.